### PR TITLE
Refactoring migration e seeder

### PR DESCRIPTION
**Pull request by msoares94**
This PR refactors User migration, makes it possible to run UserSeed multiple times without presenting a duplicate key error and adds container readiness checks

### CODE REVIEW

File: database/seeders/UserSeeder.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class UserSeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run(): void
+    {
+        $user = User::where('email', 'admin@example.com')->first();
```
Comment (WendellAdriel) [inline database/seeders/UserSeeder.php]: Add a ::query() call before the where call and put one call per line:

```php
User::query()
    ->where(...)
    ->first();
```

File: database/seeders/UserSeeder.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        if ($user) {
```
Comment (WendellAdriel) [inline database/seeders/UserSeeder.php]: Use an early return in the if so we avoid the else statement.

File: database/seeders/UserSeeder.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            $user->update([
+                'name' => 'Admin',
+                'password' => Hash::make('password'),
+                'role' => Role::ADMIN->value,
+            ]);
+        } else {
+            User::create([
```
Comment (WendellAdriel) [inline database/seeders/UserSeeder.php]: Add a ::query() call before the create call.

```php
User::query()->create([
    ...
]);
```

**Review by WendellAdriel — CHANGES_REQUESTED** [review]

**Comment by WendellAdriel** [comment]
Make sure to run the linter as well

File: database/seeders/UserSeeder.php (hunk 3 above)
Comment (msoares94) [reply database/seeders/UserSeeder.php]: Done!
```
    public function run(): void
    {
        $user = User::query()->where('email', 'admin@example.com')->first();

        if (! $user) {
            User::query()->create([
                'email' => 'admin@example.com',
                'name' => 'Admin',
                'password' => Hash::make('password'),
                'role' => Role::ADMIN->value,
            ]);

            return;
        }

        $user->update([
            'name' => 'Admin',
            'password' => Hash::make('password'),
            'role' => Role::ADMIN->value,
        ]);
    }
```

File: database/seeders/UserSeeder.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+use Modules\Auth\Support\Role;
+
+class UserSeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     */
+    public function run(): void
+    {
+        $user = User::query()->where('email', 'admin@example.com')->first();
```
Comment (WendellAdriel) [inline database/seeders/UserSeeder.php]: Put the where call and the first call in separate lines

**Review by WendellAdriel — APPROVED** [review]

**Comment by WendellAdriel** [comment]
Thanks for the contribution!

**Pull request merged**